podpac/core/algorithm/test/test_generic.py

```diff
@@ -28,80 +32,102 @@ def test_serialization(self):
 
 
 class TestArithmetic(object):
-    def setup_method(self):
-        self.settings_orig = copy.deepcopy(podpac.settings)
+    def test_init(self):
+        sine_node = SinCoords()
 
-    def teardown_method(self):
-        for key in podpac.settings:
-            podpac.settings[key] = self.settings_orig[key]
+        with podpac.settings:
+            podpac.settings.set_allow_python_eval_exec(True)
+            with warnings.catch_warnings():
+                warnings.filterwarnings("error", "Insecure evaluation.*")
+                node = Arithmetic(A=sine_node, B=sine_node, eqn="2*abs(A) - B + {offset}", params={"offset": 1})
+
+            podpac.settings.set_allow_python_eval_exec(False)
+            with pytest.warns(UserWarning, match="Insecure evaluation"):
```

Comment:
does this need a .*? 
```suggestion
            with pytest.warns(UserWarning, match="Insecure evaluation.*"):
```